src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/AbstractTextDocumentPresentationEndpointBase.cs

```diff
@@ -18,10 +18,11 @@
 using Microsoft.CodeAnalysis.Razor.Workspaces;
 using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using OmniSharp.Extensions.JsonRpc;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentPresentation
 {
-    internal abstract class AbstractTextDocumentPresentationEndpointBase<TParams> : ITextDocumentUriPresentationHandler<TParams>
+    internal abstract class AbstractTextDocumentPresentationEndpointBase<TParams> : IJsonRpcRequestHandler<TParams, WorkspaceEdit?>, IRegistrationExtension
```

Comment:
Problem number 1: I had the base class inherit from the UriPresentation handler, so it could never have worked for the TextPresentation handler